src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -470,5 +493,56 @@ public bool TryGetDateTimeOffset(out DateTimeOffset value)
             ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
             return JsonHelpers.TryParseAsISO(span, out value, out int bytesConsumed) && span.Length == bytesConsumed;
         }
+
+        /// <summary>
+        /// Reads the next JSON token value from the source and parses it to a <see cref="Guid"/>.
+        /// Returns true if the entire UTF-8 encoded token value can be successfully
+        /// parsed to a <see cref="Guid"/> value.
+        /// Returns false otherwise.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to get the value of a JSON token that is not a <see cref="JsonTokenType.String"/>.
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        public bool TryGetGuid(out Guid value)
+        {
+            if (TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedString(TokenType);
+            }
+
+            ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
```

Comment:
I just realized, we probably want escaped characters to parse successfully. For example something like:
"nnnnnnnn\u002dnnnn\u002dnnnn\u002dnnnn\u002dnnnnnnnnnnnn"

Similarly, for DateTime: ` \"\\u0030997-07-16\"` is a valid date (equivalent to `0997-07-16`).

This would apply to all `{Try}GetX()` APIs that accept JSON strings. In our case that means DateTime, DateTimeOffset, Guid.

cc @GrabYourPitchforks, @JamesNK

Reply:
Good point. Yes, I think the reader should handle escape characters for you in these helper methods.

Reply:
I've implemented escaped-character handling for (Try)GetGuid. Filing https://github.com/dotnet/corefx/issues/36202 to track the similar change for `DateTime` and `DateTimeOffset`.